api/src/main/java/org/openmrs/module/openhmis/inventory/InventoryModuleActivator.java

```diff
@@ -13,37 +13,188 @@
-	}
+	IItemDataService iItemDataService;
 
 	/**
 	 * @see BaseModuleActivator#started()
 	 */
 	@Override
+	@Autowired
+	@Transactional
 	public void started() {
 		LOG.info("OpenHMIS Inventory Module started");
+
 	}
 
 	/**
 	 * @see BaseModuleActivator#stopped()
 	 */
 	@Override
 	public void stopped() {
+
 		LOG.info("OpenHMIS Inventory Module stopped");
+
+	}
+
+	/**
+	 * @see BaseModuleActivator#contextRefreshed()
+	 */
+	@Override
+	public void contextRefreshed() {
+
+		LOG.info("OpenHMIS Inventory Module refreshed");
+		iItemDataService = Context.getService(IItemDataService.class);
+
+		processInventoryData();
+	}
+
+	private List<String> readInventoryFile() {
+		InputStreamReader is = new InputStreamReader(getClass().getClassLoader().getResourceAsStream(INVENTORY_ITEMS_FILE));
+		List<String> fileEntries = new ArrayList<>();
+		try (BufferedReader br = new BufferedReader(is)) {
+			String lineEntry;
+			while ((lineEntry = br.readLine()) != null)
+				fileEntries.add(lineEntry);
+		} catch (FileNotFoundException e) {
+			e.printStackTrace();
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		fileEntries.remove(0);
+		return fileEntries;
+	}
+
+	public void processInventoryData() {
+
+		List<String> lineEntries = readInventoryFile();
+
+		for (String lineEntry : lineEntries) {
+			Item item = createItemEntity(lineEntry.split(DATA_DELIMITER));
+			iItemDataService.save(item);
+		}
+	}
+
+	private Item createItemEntity(String[] lineValues) {
+		Item newItem = new Item();
+		try {
+			SimpleDateFormat sdf = new SimpleDateFormat(DATE_FORMAT);
+			if (StringUtils.isNotEmpty(lineValues[0])) {
+				newItem.setId(Integer.valueOf(lineValues[0]));
+			}
+			if (StringUtils.isEmpty(lineValues[1])) {
+				throw new InvalidInventoryDataException("Department value missing in input file");
+			}
+			newItem.setName(lineValues[1]);
+			newItem.setDescription(lineValues[2]);
+			newItem.setDepartment(getDepartment(lineValues[3]));
+			newItem.setDefaultPrice(getDefaultItemPrice(lineValues[4]));
+			newItem.setHasExpiration(getBooleanFlagFromString(lineValues[5]));
+			newItem.setConcept(getConcept(lineValues[6]));
+			newItem.setCreator(getUser(lineValues[7], false));
+			if (StringUtils.isEmpty(lineValues[8])) {
+				throw new InvalidInventoryDataException("Date Created value missing in input file");
+			}
+			newItem.setDateCreated(sdf.parse(lineValues[8]));
+			newItem.setChangedBy(getUser(lineValues[9], true));
+			newItem.setDateChanged(StringUtils.isEmpty(lineValues[10]) ? null : sdf.parse(lineValues[10]));
+			newItem.setRetired(getBooleanFlagFromString(lineValues[11]));
+			newItem.setRetiredBy(getUser(lineValues[12], true));
+			newItem.setDateRetired(StringUtils.isEmpty(lineValues[13]) ? null : sdf.parse(lineValues[13]));
+			newItem.setRetireReason(lineValues[14]);
+			if (StringUtils.isEmpty(lineValues[15])) {
+				throw new InvalidInventoryDataException("UUID value missing in input file");
```

Comment:
Why can't UUID be generated? This is UUID of the Item?

Reply:
@gsluthra in initializer module they are handling it as below
https://github.com/mekomsolutions/openmrs-module-initializer/blob/a8faea1d9b29d0b4c5f0bf914da8030fea859781/api/src/main/java/org/openmrs/module/initializer/api/idgen/autogen/AutoGenerationOptionsCsvParser.java#LL32C9-L32C9
Do we also need such enhancement in inventory module.

Reply:
And also if we make uuid dynamic it would create duplicate with same item name because uuid is part of unique constraint in db